syntax = "proto3";

package chat;

enum DataType {
  TEXT = 0;
  IMAGE = 1;
}

message ChatMessage {
	string user_name = 1;
	uint32 timestamp = 2;
	DataType data_type = 3;
	
	oneof content {
		string message_text = 4;
		bytes image_data = 5;
	}
	
	string image_extension = 6;
}


